Add deprecation-notice to rule CA2109 (#31797)
dotnet/docs: docs/fundamentals/code-analysis/quality-rules/ca2109.md

A public or protected event-handling method was detected.

> [!NOTE]

> > This rule last shipped with Microsoft.CodeAnalysis.Analyzers v3.3.0.

Reviewer: ```suggestion
> This rule has been deprecated. It last shipped with Microsoft.CodeAnalysis.Analyzers v3.3.0.
```

> 

> > It was removed because the threat that the analyzer warned about (an untrusted intermediary hooking a privileged event handler to a privileged event invoker) did not exist since .NET 4.5.

Reviewer: ```suggestion
> The rule was removed because the threat that the analyzer warned about (an untrusted intermediary hooking a privileged event handler to a privileged event invoker) hasn't existed since .NET Framework 4.5.
```
